### PR TITLE
[Bugfix] Fix quorum bug in blockIds report

### PR DESCRIPTION
**Pull request by frankliee**
### What changes were proposed in this pull request?
Fix a bug of report quorum.

### Why are the changes needed?
Without this patch, report quorum will fail when server number is more than replica.

### Does this PR introduce _any_ user-facing change?
No.

### How was this patch tested?
Update UTs to start more servers.

### CODE REVIEW

File: integration-test/common/src/test/java/com/tencent/rss/test/QuorumTest.java [hunk 1]
```diff
@@ -94,13 +96,20 @@ public static void initCluster() throws Exception {
...
+      new ShuffleServerInfo("127.0.0.1-20001", shuffleServers.get(0).getIp(), SHUFFLE_SERVER_PORT + 0);
     shuffleServerInfo1 =
       new ShuffleServerInfo("127.0.0.1-20002", shuffleServers.get(1).getIp(), SHUFFLE_SERVER_PORT + 1);
     shuffleServerInfo2 =
-      new ShuffleServerInfo("127.0.0.1-20003", shuffleServers.get(1).getIp(), SHUFFLE_SERVER_PORT + 2);
+      new ShuffleServerInfo("127.0.0.1-20003", shuffleServers.get(2).getIp(), SHUFFLE_SERVER_PORT + 2);
+    shuffleServerInfo3 =
+      new ShuffleServerInfo("127.0.0.1-20004", shuffleServers.get(3).getIp(), SHUFFLE_SERVER_PORT + 3);
+    shuffleServerInfo4 =
+      new ShuffleServerInfo("127.0.0.1-20005", shuffleServers.get(4).getIp(), SHUFFLE_SERVER_PORT + 4);
```
Comment (jerqi) [inline integration-test/common/src/test/java/com/tencent/rss/test/QuorumTest.java:112]: add more case,such as one partition report success, one partition report fail, but total success count reach , and cherry-pick this patch to branch 0.4.0.

Comment (frankliee) [reply integration-test/common/src/test/java/com/tencent/rss/test/QuorumTest.java:112]: ok

**Comment by jerqi** [comment]
LGTM,@sfwang218 Do you have another suggestions?

**Review by jerqi — APPROVED** [review]

**Comment by jerqi** [comment]
Don‘t forget that you should cherry-pick this patch to branch 0.4.0.

**Pull request merged**